git-branchless-navigation/src/lib.rs

```diff
@@ -544,6 +550,36 @@ pub fn switch(
                 None => return Ok(Err(ExitCode(1))),
             }
         }
+        Target::Revset(target) => {
+            let commit_sets = resolve_commits(
+                effects,
+                &repo,
+                &mut dag,
+                &[target.clone()],
+                &ResolveRevsetOptions::default(),
+            )?;
+
+            let commit_set = union_all(&commit_sets);
+            let commit_set = dag.query_heads(commit_set)?;
+            let commits = sorted_commit_set(&repo, &dag, &commit_set)?;
+
+            match commits.as_slice() {
+                [commit] => Some(CheckoutTarget::Unknown(commit.get_oid().to_string())),
+                [] => None,
```

Comment:
question: Is it intentional for you that having a revset that resolves to the empty set should do nothing? I could imagine it being useful for some workflow, although I don't have one in mind.

issue: Technically speaking, isn't the stated specification that the set must have exactly 1 head not correct?

Reply:
> Is it intentional for you that having a revset that resolves to the empty set should do nothing? 

No, I don't recall a specific intention to this. I suspect I added this to mimic the behavior of `Target::None`, but I'm also blanking on a use case. I think I'll remove this and *actually* enforce the "exactly 1 head" statement. We can relax that later if a desired workflow involving empty sets emerges. 😄